### PR TITLE
[docker] update latest Docker Python dependencies

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

Update Docker Python dependencies to commit [248521a](https://github.com/ray-project/ray/commit/248521af9ca0d68217365d3d3c9598a4142c912e)
Dependencies came from 
- [build: ray py3.9 docker (x86_64)](https://buildkite.com/ray-project/postmerge/builds/2850#018d7afa-0a06-4d0b-b66c-56017a0c223d)
- [build: ray-ml py3.9 docker (x86_64)](https://buildkite.com/ray-project/postmerge/builds/2850#018d7afa-0a0e-41cf-b32f-3092815bc91f)

## Related issue number

Closes https://www.notion.so/anyscale-hq/How-to-release-Ray-go-release-ray-28e4a2347577471288bbd7d64001fa88?pvs=4#9cc39bb28f5742778b62dd1381601e03
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by angelinalg — APPROVED** [review]

File: doc/source/ray-overview/pip_freeze_ray-ml-py39-cpu.txt [hunk 1]
```diff
@@ -62,25 +63,26 @@ build==1.0.3
 cached-property==1.5.2
 cachetools==5.3.2
 catboost==1.2.2
-certifi @ file:///croot/certifi_1690232220950/work/certifi
-cffi==1.16.0
-charset-normalizer==3.3.1
+certifi==2023.11.17
+cffi @ file:///croot/cffi_1700254295673/work
```
Comment (aslonnie) [inline doc/source/ray-overview/pip_freeze_ray-ml-py39-cpu.txt:67]: why is this changed to a local file?

**Review by aslonnie — APPROVED** [review]

File: doc/source/ray-overview/pip_freeze_ray-ml-py39-cpu.txt (hunk 1 above)
Comment (GeneDer) [reply doc/source/ray-overview/pip_freeze_ray-ml-py39-cpu.txt:67]: I'm not exactly sure why it's a local, but it came from the build doing pip freeze 😅 
This is the artifact file: https://ray-ci-artifact-branch-public.s3.amazonaws.com/248521af9ca0d68217365d3d3c9598a4142c912e/tmp/artifacts/.image-info/ray-ml:248521-py39-cpu_pip-freeze.txt
